core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -389,12 +389,6 @@ public long cleanUpAllAllocatedMemory() {
       consumers.clear();
     }
 
-    for (MemoryBlock page : pageTable) {
```

Comment:
It's not clear this code should be removed, though I agree right now it appears to be dead code because of line 382. CC @JoshRosen since I think you added this block? should line 382 just be removed?